### ISSUE TITLE
Twitter module: activation causes site-wide 500

### ISSUE THREAD

**Issue by nickmonad** [freescout-helpdesk/freescout#2145]
* Freescout version 1.8.20
* `tiredofit/docker-freescout` version 1.5.53
* PHP-fpm version 8.1.8
* Twitter module version 1.0.7

We're attempting a migration of our freescout instance to a new cluster and activating the twitter module in the new deployment causes a site-wide 500 until we manually remove the `Module/Twitter` directory and clear the cache.

```
[stacktrace]
#0 {main}
"} 
[2022-08-08 23:44:22] production.ERROR: During inheritance of JsonSerializable: Uncaught ErrorException: Return type of Tightenco\Collect\Support\LazyCollection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php:796
Stack trace:
#0 /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php(13): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8192, 'Return type of ...', '/data/Modules/T...', 796)
#1 /www/html/vendor/composer/ClassLoader.php(573): include('/data/Modules/T...')
#2 /www/html/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile('/data/Modules/T...')
#3 [internal function]: Composer\Autoload\ClassLoader->loadClass('Tightenco\\Colle...')
#4 /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/alias.php(18): class_alias('Tightenco\\Colle...', 'Illuminate\\Supp...')
#5 /data/Modules/Twitter/vendor/composer/autoload_real.php(66): require('/data/Modules/T...')
#6 /data/Modules/Twitter/vendor/composer/autoload_real.php(56): composerRequire6c4ab67ef0bf9aaf7e21c424fba9e009('caf31cc6ec7cf22...', '/data/Modules/T...')
#7 /data/Modules/Twitter/vendor/autoload.php(7): ComposerAutoloaderInit6c4ab67ef0bf9aaf7e21c424fba9e009::getLoader()
#8 /data/Modules/Twitter/Providers/TwitterServiceProvider.php(13): require_once('/data/Modules/T...')
#9 /www/html/vendor/composer/ClassLoader.php(573): include('/data/Modules/T...')
#10 /www/html/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile('/www/html/vendo...')
#11 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(279): Composer\Autoload\ClassLoader->loadClass('Modules\\Twitter...')
#12 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(189): Illuminate\Foundation\ProviderRepository->createProvider('Modules\\Twitter...')
#13 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(70): Illuminate\Foundation\ProviderRepository->compileManifest(Array)
#14 /www/html/vendor/nwidart/laravel-modules/src/Laravel/Module.php(27): Illuminate\Foundation\ProviderRepository->load(Array)
#15 /www/html/overrides/nwidart/laravel-modules/src/Module.php(279): Nwidart\Modules\Laravel\Module->registerProviders()
#16 /www/html/overrides/nwidart/laravel-modules/src/Repository.php(366): Nwidart\Modules\Module->register()
#17 /www/html/vendor/nwidart/laravel-modules/src/Providers/BootstrapServiceProvider.php(22): Nwidart\Modules\Repository->register()
#18 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(586): Nwidart\Modules\Providers\BootstrapServiceProvider->register()
#19 /www/html/vendor/nwidart/laravel-modules/src/ModulesServiceProvider.php(38): Illuminate\Foundation\Application->register(Object(Nwidart\Modules\Providers\BootstrapServiceProvider))
#20 /www/html/vendor/nwidart/laravel-modules/src/LaravelModulesServiceProvider.php(15): Nwidart\Modules\ModulesServiceProvider->registerModules()
#21 [internal function]: Nwidart\Modules\LaravelModulesServiceProvider->boot()
#22 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#23 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#24 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#25 /www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#26 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(792): Illuminate\Container\Container->call(Array)
#27 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(775): Illuminate\Foundation\Application->bootProvider(Object(Nwidart\Modules\LaravelModulesServiceProvider))
#28 [internal function]: Illuminate\Foundation\Application->Illuminate\Foundation\{closure}(Object(Nwidart\Modules\LaravelModulesServiceProvider), 23)
#29 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(776): array_walk(Array, Object(Closure))
#30 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\Foundation\Application->boot()
#31 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(213): Illuminate\Foundation\Bootstrap\BootProviders->bootstrap(Object(Illuminate\Foundation\Application))
#32 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(162): Illuminate\Foundation\Application->bootstrapWith(Array)
#33 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(146): Illuminate\Foundation\Http\Kernel->bootstrap()
#34 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#35 /www/html/public/index.php(83): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#36 {main} {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalErrorException(code: 1): During inheritance of JsonSerializable: Uncaught ErrorException: Return type of Tightenco\\Collect\\Support\\LazyCollection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php:796
Stack trace:
#0 /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php(13): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8192, 'Return type of ...', '/data/Modules/T...', 796)
#1 /www/html/vendor/composer/ClassLoader.php(573): include('/data/Modules/T...')
#2 /www/html/vendor/composer/ClassLoader.php(428): Composer\\Autoload\\includeFile('/data/Modules/T...')
#3 [internal function]: Composer\\Autoload\\ClassLoader->loadClass('Tightenco\\\\Colle...')
#4 /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/alias.php(18): class_alias('Tightenco\\\\Colle...', 'Illuminate\\\\Supp...')
#5 /data/Modules/Twitter/vendor/composer/autoload_real.php(66): require('/data/Modules/T...')
#6 /data/Modules/Twitter/vendor/composer/autoload_real.php(56): composerRequire6c4ab67ef0bf9aaf7e21c424fba9e009('caf31cc6ec7cf22...', '/data/Modules/T...')
#7 /data/Modules/Twitter/vendor/autoload.php(7): ComposerAutoloaderInit6c4ab67ef0bf9aaf7e21c424fba9e009::getLoader()
#8 /data/Modules/Twitter/Providers/TwitterServiceProvider.php(13): require_once('/data/Modules/T...')
#9 /www/html/vendor/composer/ClassLoader.php(573): include('/data/Modules/T...')
#10 /www/html/vendor/composer/ClassLoader.php(428): Composer\\Autoload\\includeFile('/www/html/vendo...')
#11 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(279): Composer\\Autoload\\ClassLoader->loadClass('Modules\\\\Twitter...')
#12 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(189): Illuminate\\Foundation\\ProviderRepository->createProvider('Modules\\\\Twitter...')
#13 /www/html/overrides/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(70): Illuminate\\Foundation\\ProviderRepository->compileManifest(Array)
#14 /www/html/vendor/nwidart/laravel-modules/src/Laravel/Module.php(27): Illuminate\\Foundation\\ProviderRepository->load(Array)
#15 /www/html/overrides/nwidart/laravel-modules/src/Module.php(279): Nwidart\\Modules\\Laravel\\Module->registerProviders()
#16 /www/html/overrides/nwidart/laravel-modules/src/Repository.php(366): Nwidart\\Modules\\Module->register()
#17 /www/html/vendor/nwidart/laravel-modules/src/Providers/BootstrapServiceProvider.php(22): Nwidart\\Modules\\Repository->register()
#18 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(586): Nwidart\\Modules\\Providers\\BootstrapServiceProvider->register()
#19 /www/html/vendor/nwidart/laravel-modules/src/ModulesServiceProvider.php(38): Illuminate\\Foundation\\Application->register(Object(Nwidart\\Modules\\Providers\\BootstrapServiceProvider))
#20 /www/html/vendor/nwidart/laravel-modules/src/LaravelModulesServiceProvider.php(15): Nwidart\\Modules\\ModulesServiceProvider->registerModules()
#21 [internal function]: Nwidart\\Modules\\LaravelModulesServiceProvider->boot()
#22 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#23 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#24 /www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#25 /www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#26 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(792): Illuminate\\Container\\Container->call(Array)
#27 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(775): Illuminate\\Foundation\\Application->bootProvider(Object(Nwidart\\Modules\\LaravelModulesServiceProvider))
#28 [internal function]: Illuminate\\Foundation\\Application->Illuminate\\Foundation\\{closure}(Object(Nwidart\\Modules\\LaravelModulesServiceProvider), 23)
#29 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(776): array_walk(Array, Object(Closure))
#30 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\\Foundation\\Application->boot()
#31 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(213): Illuminate\\Foundation\\Bootstrap\\BootProviders->bootstrap(Object(Illuminate\\Foundation\\Application))
#32 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(162): Illuminate\\Foundation\\Application->bootstrapWith(Array)
#33 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(146): Illuminate\\Foundation\\Http\\Kernel->bootstrap()
#34 /www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#35 /www/html/public/index.php(83): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#36 {main} at /data/Modules/Twitter/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php:13)
```

Seems to be related to JSON serialization, but not sure if it's due to an incorrect version of PHP, the Twitter module, or something else. Any advice or insight would be greatly appreciated! Thanks!

**Comment by zachherbert** [comment]
Freescout team, could we please get an update on this? We are using Freescout in production and have spent a lot of money on your plugins. Thanks.

**Comment by freescout-helpdesk** [comment]
Fixed in Twitter Module v1.0.8

**Comment by nickmonad** [comment]
Thank you!
